cloud/client.go

```diff
@@ -183,27 +183,33 @@ func (c *client) doCall(ctx context.Context, method, url string, opts ...request
 
 	var status int
 	var body string
-	var err error
+	var callErr error
 	duration := time.Millisecond * 100
 	for attempt := 0; attempt < maxAttempt; attempt++ {
-		status, body, err = c.doCallImp(ctx, r, method, url, opts...)
-
-		if err != nil && strings.Contains(err.Error(), "context canceled") {
+		status, body, callErr = c.doCallImp(ctx, r, method, url, opts...)
+		if !shouldRetry(status, body, callErr, c.warnFunc) {
```

Comment:
I think by moving this call further up, we are printing the warning "retrying HTTP request" when the error is context cancelled but not actually an instance of it according to the  `errors.Cause` check that was added in `shouldRetry`.  I pasted a screenshot of this in a different comment